int32 instr_head
int32 read_head
int32 write_head
Stack[] stacks
int32 active_stack
string reg_AX
string reg_BX
string reg_CX
string[] memory
string[] environment_buffer
string[] input_buffer
string output
string current_instruction
